### PR TITLE
Bugfix/startup walk back

### PR DESCRIPTION
**Pull request by n-hutton**
This PR does two things:

- Fixes overly strict requirement on the merkle state matching the chain state
- Creates a whiltelist of initial miners

### CODE REVIEW

File: libs/constellation/src/constellation.cpp [hunk 1]
```diff
@@ -526,24 +531,21 @@ bool Constellation::OnBringUpExternalNetwork(
...
+  else
+  {
+    FETCH_LOG_INFO(LOGGING_NAME, "No snapshot to reset consensus with.");
   }
 
-  execution_manager_ = std::make_shared<ExecutionManager>(
-      cfg_.num_executors, cfg_.log2_num_lanes, storage_,
-      [this] { return std::make_shared<Executor>(storage_); }, tx_status_cache_);
+  // Update with genesis to trigger loading any saved state
+  consensus_->UpdateCurrentBlock(*chain_->CreateGenesisBlock());
```
Comment (ejfitzgerald) [inline libs/constellation/src/constellation.cpp:548]: This doesn't seem right?

File: libs/constellation/src/constellation.cpp [hunk 2]
```diff
@@ -927,39 +929,69 @@ bool Constellation::GenesisSanityChecks(GenesisFileCreator::Result genesis_statu
...
+      FETCH_LOG_WARN(LOGGING_NAME, "The revert operation failed!");
       return false;
     }
 
-    if (!storage_->RevertToHash(current_block->merkle_hash, current_block->block_number))
+    FETCH_LOG_INFO(LOGGING_NAME, "Reverted storage unit.");
+    FETCH_LOG_INFO(LOGGING_NAME, "Reverting DAG to: ", current_block->block_number);
+
+    // Need to revert the DAG too
+    if (dag_ && !dag_->RevertToEpoch(current_block->block_number))
```
Comment (ejfitzgerald) [inline libs/constellation/src/constellation.cpp:988]: 👍

File: libs/constellation/src/constellation.cpp (hunk 1 above)
Comment (ejfitzgerald) [reply libs/constellation/src/constellation.cpp:548]: This should be the heaviest block on the chain presumably?

Comment (n-hutton) [reply libs/constellation/src/constellation.cpp:548]: I think it is fine - you are allowed to update consensus with a current block that is not contigous in block number so while this looks weird I think it is safe. I did previously have a solution that updated it to the current block that we recovered to in constellation but I deemed it unneccessary code.

**Review by ejfitzgerald — APPROVED** [review]

**Pull request merged**